### DISCUSSION TITLE
ÄNDERUNG: Skript zum Herunterladen von Buch-Listings

### DISCUSSION THREAD

**Discussion by TobiasPSP** [TobiasPSP/OReilly#11] [WICHTIG]
In **Kapitel 1** wird beschrieben, wie Sie mit dem folgenden Code den Befehl `L` zum Herunterladen von Buchskripten einrichten:

```powershell
Invoke-RestMethod -Uri https://tinyurl.com/codeAusBuch | 
New-Item -Path function: -Name L -Force | 
Out-Null
```

Leider hat *tinyurl.com* ohne Ankündigung gestern (19.11.21) einen Browsercheck zwischengeschaltet, der dafür sorgt, dass es zu Fehlermeldungen kommt und der Befehl `L` nicht mehr angelegt werden kann.

Um das Problem zu beheben, verzichten Sie bitte auf *tinyurl.com*  und geben stattdessen die direkte Url des Skripts an:

```powershell
Invoke-RestMethod -Uri https://raw.githubusercontent.com/TobiasPSP/OReilly/main/PowerShell/loadScript.ps1 | 
New-Item -Path function: -Name L -Force | 
Out-Null
```

Anschließend steht Ihnen wieder wie gewohnt der Befehl `L` zur Verfügung, mit dem Sie die Buch-Listings unter Angabe ihrer Listingnummer herunterladen können. Beispiel:

```powershell
L 1.1
```

**Comment by MartinWagener** [comment]
Hallo,
bei mir geht tinyurl noch.

**Comment by Federahannes** [comment]
Ich habe im October 2022 die erst Auflage (2021) gekauft.
Gestern/heute (28.12.2022) habe ich die Listings mit der Tinyurl (Listings L 1.13 - L 1.16) ohne Probleme implementieren können.

Allerdings gibt es inzwischen offenbar neuere PS Versionen (7.3.1) und auch neuere Versionen von VSCode. Die im Buch beschriebenen Anpassungen über JASON Skripte (bspw. zur opaken Darstellung mit Acrylic sind so nicht mehr durchführbar bzw. obsolete, da in den Einstellungen konfigurierbar.

Auch die beschriebenen Schritte zur Einstellung der PS Version in VSCode ist nur mir Abstrichen nachvollziehbar, da man auch dies nun über das Konfigurations-Menüe vornimmt

**Comment by TobiasPSP** [comment]
Danke für das Update. Bei gedruckten Büchern ist es leider unvermeidlich - speziell bei sich so schnell ändernder Software wie VSCode - dass sich nach Drucklegung einige Details weiterentwickeln und verändern. Umso hilfreicher sind Deine Rückmeldungen, die ich für die nächste Auflage berücksichtige. 

Tatsächlich sind immer mehr Einstellungen in VSCode nun auch über die grafischen Dialoge möglich. Die Einstellungen selbst werden dabei aber immer in die JSON-Dateien eingetragen bzw. von dort gelesen. Die Konfiguration direkt über die JSON-Dateien sollte daher eigentlich nach wie vor in jedem Fall funktionieren. 

Gibt es JSON-Einstellungen, die keine Wirkung (mehr) zeigen? 

Falls ja, kann dies an mehreren (konkurrierenden) JSON-Dateien liegen. VSCode unterstützt mehrere JSON-Dateien (eine für alle User, eine für den aktuellen User direkt im Userprofil, sowie projektbasierte JSON-Dateien direkt in dem Projektordner, den der Explorer gerade anzeigt. Die jeweils spezifischste JSON-Einstellung gewinnt (bei konkurrierenden Einstellungen).

Wenn man also eine JSON-Datei direkt im aktuellen Projektordner angelegt hat (und dies kann auch transparent passieren, wenn man die neuen grafischen Dialoge einsetzt), haben diese Vorrang vor den übrigen JSON-Dateien.

**Reply by Federahannes** [reply]
Gerne, das war die Idee der Rückmeldung.. und danke für die Erläuterungen bezüglich JASON. 

Der Laptop ist vom Bildungsträger gemanged. Die ExecutionPolicies hatte ich überall (VSC, PS, WPS) für dies Scopes Process, CurrentUser und LocalMachine jeweils auf ByPass gesetzt, aber Process geht nach jedem Neustart zurück auf Undefined. 
Im Gegensatz zur Darstellung in Ihrem Buch war die betreffende Zeile für "use Acrylic" so ziemlich ganz am Schluss (nach den ganzen geklammerten Definitionen und vebor die letzten Klammern geschlossen wurden.. .. dort hatte ich Ihre Zeilen eingefügt (inkl. jener für opcity)). Dies hatte allerdings keine Auswirkung. 

Übrigens das mit dem GitDesktop bzw. das Herunterladen des betreffenden Repositories war auch etwas anders als im Buch beschrieben.

Andererseits stehe ich mit dem VSCode sowieso auf Kriegsfuß, bspw. hatte ich mit dem Beep-Befehl (S. 81 im Buch) herumgespielt. Hatte zunächst auch geklappt, aber nach 3-4 "Erweiterungen" (Aneinanderreihungen des Befehls mit "wait-event" dazwischen kam dann der übliche Protest, irgendwas mit JASON-File s. Screenshot anbei)
![JASON-Fehlermeldung bei Ausführung von Codebeispiel](https://user-images.githubusercontent.com/110595225/210055441-2d575e8b-e671-4d2a-a5d8-6833263defb1.PNG)
![launch JASON - add config](https://user-images.githubusercontent.com/110595225/210055904-3a6d2013-c2b4-4426-be63-1ed8f4ed62c8.PNG)
